### ISSUE TITLE
GPU-acceleration of inference

### ISSUE THREAD

**Issue by Hyrtsi** [serengil/deepface#517]
Greetings.

Question 1: Is GPU acceleration for inference supported currently?
Question 2: If not, are there any plans for that?

If it does not exist, I could look at the code and try to implement it myself and naturally, open a PR. I think it would be easy to implement.

I'm interested in running this code on mobile devices really fast. I will be testing the models with Nvidia devices and possibly converting the model to .onnx and to TensorRT format. My output would be the inference speed versus accuracy on several platforms, models and quantization levels. If someone has any input to this idea I would be happy to hear.

Thank you for the amazing repository and your good work,
Eljas (@hyrtsi)

**Comment by serengil** [comment]
deepface uses tensorflow and keras pair in the background. so, yes you can use gpu. 

you need to install tensorflow-gpu package first and then install the deepface.

**Comment by Hyrtsi** [comment]
Ok, thank you for your fast response :)

**Comment by haseebkhawaja** [comment]
> deepface uses tensorflow and keras pair in the background. so, yes you can use gpu.
> 
> you need to install tensorflow-gpu package first and then install the deepface.

i think tensorflow-gpu package is not enough for fast inference... world is moving towards edge devices... we have the faster inference with low resources
